Source/Fields/LMCCylindricalCavity.cc

```diff
@@ -152,9 +173,9 @@ namespace locust
     	double jl_of_k1r_by_k1r = 1./(2.*l) * (boost::math::cyl_bessel_j(l-1, k1*r) + boost::math::cyl_bessel_j(l+1, k1*r));
     	double jPrime = 1./2. * boost::math::cyl_bessel_j(l-1, k1*r) - boost::math::cyl_bessel_j(l+1, k1*r);
     	TE_H.push_back(-k3/k1 * jPrime * cos(l*theta) * cos(k3*z));
-    	TE_H.push_back(-l*k3/k1 * jl_of_k1r_by_k1r * sin(l*theta) * cos(k3*z));
     	TE_H.push_back(boost::math::cyl_bessel_j(l, k1*r) * cos(l*theta) * sin(k3*z));
-        return TE_H;
+    	TE_H.push_back(-l*k3/k1 * jl_of_k1r_by_k1r * sin(l*theta) * cos(k3*z));
+        return TE_H; // r, z, theta
```

Comment:
Sorry, is there a specific reason you're changing the ordering of the coordinates here (and in TM_E)? Was there an issue somewhere else that I'm missing? I only see this implemented in print statements like in new line 257 of `LMCCavitySignalGenerator.cc`

Reply:
Right, this was done to allow efficient access to `r` and `theta` components of fields with e.g. `tEr = anE_normalized.front()` and  `tEtheta = anE_normalized.back()` , whether the `z` component is present or not.  This was found to help with the recently-implemented TM calculations where the `E` field has 3 components instead of 2.  We could probably accomplish this some other way, but this seemed efficient at the time.